### PR TITLE
BOAC-5364: refactors attachments component, fixes bugs

### PR DESCRIPTION
**Pull request by lyttam**
https://jira-secure.berkeley.edu/browse/BOAC-5364

I made AdvisingNoteAttachments reusable inside AdvisingNote.

### CODE REVIEW

File: src/components/note/AdvisingNote.vue [hunk 1]
```diff
@@ -141,99 +141,34 @@
       :function-confirm="confirmedRemoveAttachment"
       modal-header="Delete Attachment"
     >
-      Are you sure you want to delete the <strong>'{{ displayName(existingAttachments, deleteAttachmentIndex) }}'</strong> attachment?
+      Are you sure you want to delete the <strong>'{{ displayName(props.note.attachments, deleteAttachmentIndex) }}'</strong> attachment?
```
Comment (johncrossman) [inline src/components/note/AdvisingNote.vue]: @lyttam - In template code, `props` is not necessary. You should be able to just use `note.attachments`

File: src/components/note/AdvisingNote.vue [hunk 2]
```diff
@@ -141,99 +141,34 @@
...
-      </div>
-    </div>
+    <AdvisingNoteAttachments
+      v-if="isOpen && !note.legacySource && currentUser.uid === author.uid"
+      :add-attachments="addNoteAttachments"
+      class="attachments-edit pt-3"
+      :disabled="!!(isUpdatingAttachments || noteStore.boaSessionExpired)"
+      downloadable
+      :id-prefix="`note-${props.note.id}-`"
+      :note="props.note"
```
Comment (johncrossman) [inline src/components/note/AdvisingNote.vue]: Same here.

**Comment by johncrossman** [comment]
Congrats on the deletion of the Notes mixin! All looks good. I'll wait for your thoughts on the `props` usage.

File: src/components/note/AdvisingNote.vue (hunk 1 above)
Comment (lyttam) [reply src/components/note/AdvisingNote.vue]: @johncrossman thanks, will amend.

**Pull request merged**